test/runs/test_set_project.py

```diff
@@ -16,7 +18,7 @@ def test_set_project_with_global_run():
 
 def test_set_project_without_run():
   with mock.patch("sigopt.get_run_id", mock.Mock(return_value=None)):
-    with pytest.warns(None) as warnings:
+    with warnings.catch_warnings():
+      warnings.simplefilter("error")
       set_project("test-123")
-  assert len(warnings) == 0
```

Comment:
Are these equivalent?

Reply:
https://stackoverflow.com/questions/45671803/how-to-use-pytest-to-assert-no-warning-is-raised said that this catch_warnings() call was the pytest >7 way to confirm that no warnings were issued (which is what this code was trying to do). 

The simplefilter("error") turns all warnings into errors, and the catch_warnings() forces it to handle them. The warns(None) stuff was deprecated in pytest7 and now is gone in pytest8.